src/com/xilinx/rapidwright/design/DesignTools.java

```diff
@@ -3385,74 +3385,77 @@ public static void createCeSrRstPinsToVCC(Design design) {
+                            if (net != null) {
                                 if (belPinName == CE) {
-                                    // CE
-                                    assert(belPinNet.isVCCNet());
+                                    if (!net.isVCCNet()) {
+                                        continue;
+                                    }
+                                    // CE: it is possible for sitewire to be assigned to a non VCC net, but a SitePinInst to not yet exist
                                 } else {
-                                    // SR
-                                    assert(belPinNet.isStaticNet());
+                                    assert(belPinName == SR);
+                                    if (!net.isStaticNet()) {
+                                        continue;
+                                    }
+                                    // SR: it is possible for sitewire to be assigned the GND net, yet still be routed to VCC
                                 }
                             }
-                        }
 
-                        if (spi != null) {
-                            assert(belPinName == SR);
-                            // Move the SR pin from GND to VCC
-                            spi.setNet(vccNet);
-                        } else {
-                            spi = new SitePinInst(false, sitePinName, si);
+                            if (assertionsEnabled) {
+                                BELPin belPin = bel.getPin(belPinName);
+                                Net belPinNet = si.getNetFromSiteWire(belPin.getSiteWireName());
+                                if (belPinNet != null) {
+                                    if (belPinName == CE) {
+                                        // CE
+                                        assert(belPinNet.isVCCNet());
+                                    } else {
+                                        // SR
+                                        assert(belPinNet.isStaticNet());
+                                    }
+                                }
+                            }
+
+                            if (spi != null) {
+                                assert(belPinName == SR);
+                                // Move the SR pin from GND to VCC
+                                spi.setNet(vccNet);
+                            } else {
+                                spi = new SitePinInst(false, sitePinName, si);
+                            }
+                            boolean updateSiteRouting = false;
+                            vccNet.addPin(spi, updateSiteRouting);
                         }
-                        boolean updateSiteRouting = false;
-                        vccNet.addPin(spi, updateSiteRouting);
+                    }
+                } else if (si.getSiteTypeEnum() == SiteTypeEnum.MMCM) {
+                    for (String sitePinName : new String[]{"CLKFBIN", "CLKIN2"}) {
```

Comment:
The MMCM in both Versal and UltraScale+ have the same set of site pins.  Is there something we should also be doing for UltraScale+?